apps/endatix-hub/app/forms/[formId]/page.tsx

```diff
@@ -0,0 +1,40 @@
+import dynamic from "next/dynamic";
+import { Form, FormDefinition } from "../../../types";
+import { getFormById, getFormDefinitionByFormId } from "@/services/api";
+
+const SurveyCreatorComponent = dynamic(() => import('@/components/survey-creator'), {
+  ssr: false,
+});
+
+export default async function FormCreator({ params }: { params: { formId: string } }) {
+  const { formId } = params;
+
+  let form: Form | null = null;
+  let formJson: Object | null = null;
+
+  try {
+    form = await getFormById(formId);
+
+    const response: FormDefinition = await getFormDefinitionByFormId(formId);
+    formJson = response?.jsonData ? JSON.parse(response.jsonData) : null;
+  } catch (error) {
+    console.error("Failed to load form:", error);
+    formJson = null;
+  }
+
+  if (!form || !formJson) {
+    return <div>Form not found</div>;
+  }
+
+  return (
+    <div className="flex min-h-screen flex-col items-center p-8">
```

Comment:
@dspetrov  - What about just pass FormProps object instead of decomposed props? Makes code cleaner and intent clear. Do you agree with this approach?

Reply:
I agree and made the change.